File: app/src/main/java/com/sliebald/cula/MyApplication.java
package com.sliebald.cula;

import android.app.Application;

/**
 * Application class to provide access to the application context for e.g. the Database.
 * Based on https://stackoverflow.com/questions/21818905/get-application-context-from-non
 * -activity-singleton-class
 */
public class MyApplication extends Application {

    private static MyApplication mContext;

    /**
     * Get the Application context of the App.
     *
     * @return The {@link android.content.Context} of this app.
     */
    public static MyApplication getContext() {
        return mContext;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = this;
    }
}

File: app/src/main/java/com/sliebald/cula/MyApplication.kt
package com.sliebald.cula

import android.app.Application

/**
 * Application class to provide access to the application context for e.g. the Database.
 * Based on https://stackoverflow.com/questions/21818905/get-application-context-from-non
 * -activity-singleton-class
 */
class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        context = this
    }

    companion object {
        /**
         * Get the Application context of the App.
         *
         * @return The [android.content.Context] of this app.
         */
        lateinit var context: MyApplication
            private set
    }
}